.github/workflows/quality.yml

```diff
@@ -23,12 +24,56 @@ jobs:
         run: |
           ./gradlew ktlintCheck --continue
       - name: Upload reports
+        if: always()
         uses: actions/upload-artifact@v2
         with:
           name: ktlinting-report
-          path: vector/build/reports/ktlint/*.*
+          path: |
+            */build/reports/ktlint/ktlint*/ktlint*.txt
+      - name: Handle Results
+        if: always()
+        id: get-comment-body
+        run: |
+          results="$(cat */*/build/reports/ktlint/ktlint*/ktlint*.txt */build/reports/ktlint/ktlint*/ktlint*.txt | sed -r "s/\x1B\[([0-9]{1,3}(;[0-9]{1,2})?)?[mGK]//g")"
+          if [ -z  "$results" ]; then
+            body="👍 ✅ 👍"
+          else
+            body="\`👎 ❌ 👎 ${results}\`"
+          fi
+          body="${body//'%'/'%25'}"
+          body="${body//$'\n'/'%0A'}"
+          body="${body//$'\r'/'%0D'}"
+          body="$( echo $body | sed 's/\/home\/runner\/work\/element-android\/element-android\//\`<br\/>\`/g')"
+          body="$( echo $body | sed 's/\/src\/main\/java\// 🔸 /g')"
+          body="$( echo $body | sed 's/im\/vector\/app\///g')"
+          body="$( echo $body | sed 's/im\/vector\/lib\/attachmentviewer\///g')"
+          body="$( echo $body | sed 's/im\/vector\/lib\/multipicker\///g')"
+          body="$( echo $body | sed 's/im\/vector\/lib\///g')"
+          body="$( echo $body | sed 's/org\/matrix\/android\/sdk\///g')"
+          body="$( echo $body | sed 's/\/src\/androidTest\/java\// 🔸 /g')"
+          echo "::set-output name=body::$body"
+      - name: Find Comment
+        if: always()
+        uses: peter-evans/find-comment@v1
+        id: fc
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          comment-author: 'github-actions[bot]'
+          body-includes: Ktlint Results
+      - name: Publish ktlint results to PR
+        if: always()
```

Comment:
OOI, is there a necessity to add this line?

Reply:
Totally, while when `ktlint` fails thats when we should publish the results, also that was the fix for the artifacts. The problem was that the jobs was killed so nothing happened after that

Reply:
Ok, understood, thanks!